File: 205-isomorphic-strings/205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hash_map = {}
        i = 0
        while (i < len(s)):
            if s[i] not in hash_map:
                if t[i] in hash_map.values():
                    return False
                else:
                    hash_map[s[i]] = t[i]
            else:
                if hash_map[s[i]] != t[i]:
                    return False
            i += 1
        return True

File: 205-isomorphic-strings/205-isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const isIsomorphic = function(s, t) {
  const map = new Map();
  const set = new Set();
  for (let i = 0; i < s.length; i++) {
    let sChar = s[i];
    let tChar = t[i];
    if (map.has(sChar)) {
      if (map.get(sChar) !== tChar) return false;
    } else {
      if (set.has(tChar)) return false;
      map.set(sChar, tChar);
      set.add(tChar);
    }
  }
  return true;
};